lib/cron.dart

```diff
@@ -47,6 +47,9 @@ class Schedule {
   /// The weekdays a Task should be started.
   final List<int>? weekdays;
 
+  /// The datetime a Task last run.
+  DateTime lastTime = DateTime(0, 0, 0, 0, 0, 0, 0);
```

Comment:
Thinking about it more, I think this field shouldn't be here, rather it should be moved to `_ScheduledTask`.
The check should be in
```
  void tick(DateTime now) {
    if (_closed) return;
    if (!schedule.shouldRunAt(now)) return;
// this is the new part:
   if (<big-condition>) {
    return;
   }
    _run();
  }
``` 

The reason being: Schedule instances may be shared even between `Cron` instances, and we could get parallel updates. `_ScheduledTask` is private to the `Cron` instance and it is not shared, it would be the correct place to put this field in.